docs/src/rfc/text/0000-boot-manager.md

```diff
@@ -0,0 +1,707 @@
+# RFC: Boot Manager
+
+This RFC proposes a modular Boot Manager for Patina as a drop-in replacement for EDK2's BdsDxe driver. The design
+transforms the monolithic 1200-line EDK2 BdsDxe into a service-oriented architecture with 3 required core services
+and 9 optional services, leveraging Rust's type system for compile-time safety guarantees. The orchestrator
+coordinates boot sequencing using dependency injection, replacing C function pointers and global state with type-safe
+trait implementations. The redesign achieves significant binary size reduction while eliminating entire classes of
+vulnerabilities through Rust's ownership model.
+
+## Change Log
+
+- 2025-11-04: Initial RFC draft
+
+## Motivation
+
+The current EDK2 Boot Manager (BDS - Boot Device Selection) implementation represents a monolithic architecture that
+has accumulated technical debt over nearly two decades. Originally designed for UEFI 2.0 (2006), it carries legacy
+features deprecated for 19+ years while mixing multiple responsibilities in a single 1200-line driver. This creates
+maintenance burdens, security vulnerabilities, and platform integration complexity that modern Rust-based firmware can
+eliminate.
+
+**Current Problems:**
+
+1. **Monolithic Architecture**: Single driver handles multiple distinct responsibilities from console management to
+   boot execution, variable management, hotkeys, and platform integration
+2. **Memory Safety Vulnerabilities**: Manual pointer management, unchecked array access, and type confusion risks
+3. **Legacy Baggage**: Lang/LangCodes variables deprecated since UEFI 2.0, Option ROM dispatch security risks,
+   CSM/Legacy BIOS support bloat
+4. **Testing Difficulties**: Global state and tight coupling make unit testing nearly impossible
+5. **Platform Customization Complexity**: Library override patterns require rebuilding entire firmware stacks
+6. **Binary Bloat**: All features compiled in whether needed or not
+
+**Compatibility Definition** (consistent with other Patina RFCs):
+
+1. **Functional Equivalence**: The Rust implementation provides all functionality required by the UEFI Boot Manager
+2. **Semantic Preservation**: Boot sequence, variable handling, and platform callbacks maintain identical behavior
+3. **UEFI Compliance**: Full adherence to UEFI 2.11 and PI 1.9 specifications (December 2024)
+4. **Platform Compatibility**: Supports all 15+ EDK2 platforms currently using BdsDxe with optional feature customization
```

Comment:
Many many more platforms use EDK II than 15 (that may be the number in edk2-platforms). This line probably can be changed to something like "Supports platform extensibility to meet current scenarios"